crates/services/consensus_module/poa/src/signer.rs

```diff
@@ -173,4 +208,51 @@ mod tests {
             .is_available())
         };
     }
+
+    // The tests are against a keypair generated using fuel-core-keygen, which has
+    // been tweaked to display the public key associated with a keypair.
+    // The keypair used in these tests is the following:
+    // {
+    //    "address:"e2f3f5109c56eec359c124cbf25f35dcc1495b0bbdac5848e0ff37a86fe69a6d",
+    //    "public_key":"9ab6229de634056cdc67dfba26e6a06e4ba082693ea30395e5994b113ab6c6e3189a12a10d8fb08d1d28f7117ca34f6b16c5132acd9570de6e7a005f6bbd8f3d",
+    //    "secret":"2708b7bad8b5b52d031e5795c1d1995660185f464900cbd593328eb433bdb7f6","type":"block-production"
+    // }
+    #[test]
+    fn public_key() {
+        assert_eq!(SignMode::Unavailable.public_key().unwrap(), None);
+
+        let secret_key = SecretKey::from_str(
+            "2708b7bad8b5b52d031e5795c1d1995660185f464900cbd593328eb433bdb7f6",
+        )
+        .expect("Secret key construction should not fail");
+
+        let public_key = "9ab6229de634056cdc67dfba26e6a06e4ba082693ea30395e5994b113ab6c6e3189a12a10d8fb08d1d28f7117ca34f6b16c5132acd9570de6e7a005f6bbd8f3d";
```

Comment:
trying to construct a public key from this string slice fails, I am not sure why that is the case. 

```rust
        let _pk = PublicKey::from_str(public_key).unwrap();
```

yields an error `InvalidPublicKey`. 

If I instead convert the derived public key to a string literal, and i compare it with the string literal above, the test passes.